crypto/bio/bio_test.cc

```diff
@@ -591,6 +592,38 @@ TEST(BIOTest, Gets) {
   EXPECT_EQ(c, 'a');
 }
 
+TEST(BIOTest, ExternalData) {
+  // Define a custom data structure
+  typedef struct {
+    int custom_data;
+  } CustomData;
+
+  struct customdata_deleter {
```

Comment:
You could just pass a callback to the appropriate memory free function to `BIO_get_ex_new_index` instead of all this? But it also works as-is now.

Reply:
Sure just applied it :)